panc/src/main/java/org/quattor/pan/dml/functions/Match.java

```diff
@@ -53,9 +53,9 @@ public static Operation getInstance(SourceRange sourceRange,
 
 		// Optimize the operation if compile time constants are given for the
 		// regular expression and match flags.
-		if (operations.length > 0
-				&& operations[1] instanceof StringProperty
-				&& (operations.length < 3 || operations[2] instanceof StringProperty)) {
+		if (operations.length == 2
```

Comment:
this changes the code more than the title suggest. 
to fix the title, i think it suffices to set `operations.length >=2`
the current patch ignores flags for static matches, and requires that also the value to search (and not only the pattern and optional flags) are "constants/strings"

Reply:
Ok, respin...